src/main/java/bridge/views/InputView.java

```diff
@@ -0,0 +1,58 @@
+package bridge.views;
+
+import camp.nextstep.edu.missionutils.Console;
+
+import java.util.List;
+
+public final class InputView {
+
+    static final String UP_COMMAND = "U";
+    static final String DOWN_COMMAND = "D";
+    public static final String QUIT_COMMAND = "Q";
+    public static final String RESTART_COMMAND = "R";
+
+    private final OutputView outputView;
+
+    public InputView() {
+        outputView = new OutputView();
+    }
+
+    public int readBridgeSize() {
+        outputView.printInputBridgeSizeText();
+        String readBridgeSize = Console.readLine();
+
+        try {
+            InputChecker.validateNumeric(this, readBridgeSize);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            readBridgeSize = String.valueOf(readBridgeSize());
+        }
+        return Integer.parseInt(readBridgeSize);
+    }
+
+    public String readMoving() {
+        outputView.printInputMoveDirection();
+        String readDirection = Console.readLine();
+
+        try {
+            InputChecker.validateCommand(List.of(UP_COMMAND, DOWN_COMMAND), readDirection);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            readDirection = readMoving();
+        }
+        return readDirection;
+    }
+
+    public String readGameCommand() {
+        outputView.printRestartQuestion();
+        String readCommand = Console.readLine();
+
+        try {
+            InputChecker.validateCommand(List.of(QUIT_COMMAND, RESTART_COMMAND), readCommand);
+        } catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+            readCommand = readGameCommand();
+        }
+        return readCommand;
+    }
```

Comment:
```suggestion
    public String readGameCommand() {
        outputView.printRestartQuestion();
        String readCommand = Console.readLine();

        try {
            InputChecker.validateCommand(List.of(QUIT_COMMAND, RESTART_COMMAND), readCommand);
            return readCommand;
        } catch (IllegalArgumentException e) {
            System.out.println(e.getMessage());
            return readGameCommand();
        }
    }
```
1. 위와 같이 리턴을 하도록 바꿀 수 있을 것 같습니다.
2. System.out.println이 컨트롤러에도 존재하고 InputView에도 존재하네요. `System.out#println`은 출력의 구현 중 한 방법이라고 생각되는데 OutputView에만 존재하도록 하면 더 좋지 않을까요? 어떤 장단점이 있을까요?

Reply:
1. 
함수의 리턴 타입은 무조건 밖에 마지막 블록에 있어야 하는건줄 알았는데 
try 블록과 catch 블록에 하나씩 있으면 가능한지 처음 알았네요.

덕분에 많이 알아갑니다.

2.
처음에는 domain에서 예외처리를 하다보니 
당연히 view와 연관되면 안되기 때문에 출력을 따로 했는데,

생각해보니 충분히 가능하네요.

해당 방식의 장점은 모든 출력 관련된 문구는 OutputView class에 있으므로
타인이 코드의 위치를 찾기 편해지고, 수정도 편해집니다.

단점은 음 굳이.. OutputView의 의존성이 높아지는 것 정도..?